cmd/plume/prerelease.go

```diff
@@ -575,24 +741,44 @@ func awsUploadAmiLists(ctx context.Context, bucket *storage.Bucket, spec *channe
 // This includes uploading the ami_vmdk image to an S3 bucket in each EC2
 // partition, creating HVM and PV AMIs, and replicating the AMIs to each
 // region.
-func awsPreRelease(ctx context.Context, client *http.Client, src *storage.Bucket, spec *channelSpec, imageInfo *imageInfo) error {
+func awsPreRelease(ctx context.Context, system string, client *http.Client, src *storage.Bucket, spec *channelSpec, imageInfo *imageInfo, imageMetadata *map[string]interface{}) error {
 	if spec.AWS.Image == "" {
 		plog.Notice("AWS image creation disabled.")
 		return nil
 	}
 
-	imageName := fmt.Sprintf("%v-%v-%v", spec.AWS.BaseName, specChannel, specVersion)
-	imageName = regexp.MustCompile(`[^A-Za-z0-9()\\./_-]`).ReplaceAllLiteralString(imageName, "_")
-	imageDescription := fmt.Sprintf("%v %v %v", spec.AWS.BaseDescription, specChannel, specVersion)
+	imageData, err := getSpecAWSImageName(spec, imageMetadata)
+	if err != nil {
+		return fmt.Errorf("Could not generate the image filname: %v", err)
+	}
 
-	imagePath, err := getImageFile(client, src, spec.AWS.Image)
+	imageFileName := imageData["imageFileName"]
+	imageName := imageData["imageName"]
+	imageDescription := imageData["imageDescription"]
+
+	imagePath, err := getImageFile(system, client, spec, src, imageFileName)
 	if err != nil {
 		return err
 	}
 
+	if system == "fedora" {
+		vmdkImagePath := strings.Replace(imagePath, "raw", "vmdk", 1)
+		if _, err := os.Stat(vmdkImagePath); err == nil {
+			plog.Printf("Reusing existing vmdk image %q", vmdkImagePath)
+			return nil
+		}
+
+		cmd := exec.Command("qemu-img", "convert", "-f", "raw", "-O", "vmdk", "-o", "adapter_type=lsilogic,subformat=streamOptimized,compat6", imagePath, vmdkImagePath)
```

Comment:
I'm not sure how I feel about plume doing this convert. We should probably either download existing bits that are in the proper format or have some other tool/script, outside of plume itself, that downloads & preps the correct release bits.

cc @bgilbert

Reply:
cc @dustymabe

Reply:
If we don't feel comfortable doing the convert here then I think it would be OK to do the convert in the fedimg glue code that we will still need. @arithx do you see any other need ever for converting the input disk to the required format for upload?

Reply:
@dustymabe not inside of plume. Plume is built around taking already built release artifacts and publishing them. They should already be built and simply be consumed.

Reply:
@sayanchowdhury can we add this to the fedimg glue code?

Reply:
after a brief discussion in IRC we identified our options as:

1 - upload the raw (takes longer and wastes bandwidth)
2 - download the raw.xz uncomporess and convert to vmdk outside of plume
3 - do the convert inside plume (against philosophy) 
4 - make fedora produce a vmdk

let's go with 1 and I'll see about talking to releng to see if we can do 4

Reply:
for `4` I opened this PR to see if we can make progress there: https://pagure.io/pungi-fedora/pull-request/683

Reply:
It looks like @dustymabe's PR has merged, can you please update this to remove the VMDK creation and use the VMDK images produced by Fedora rather than converting the raw image in this code.

Reply:
@sayanchowdhury you can try with the vmdk from https://koji.fedoraproject.org/koji/taskinfo?taskID=32866399. For some reason the vmdk isn't getting copied to the output directory yet. Opened an issue for that: https://pagure.io/releng/issue/8148